src/esp/assets/ResourceManager.cpp

```diff
@@ -58,6 +63,231 @@ bool ResourceManager::loadScene(const AssetInfo& info,
   }
 }
 
+bool ResourceManager::loadPhysicalScene(
+    const AssetInfo& info,
+    PhysicsManager& _physicsManager,
+    scene::SceneNode* parent /* = nullptr */,
+    bool attach_physics, /* = false */
+    DrawableGroup* drawables /* = nullptr */) {
+  bool meshSuccess = loadScene(info, parent, drawables);
+
+  if (attach_physics) {
+    physics::BulletRigidObject* physNode = new physics::BulletRigidObject(parent);
+
+    LOG(INFO) << "Physics node " << physNode;
+    if (physNode == nullptr) {
+      LOG(ERROR) << "Cannot instantiate physics node.";
+      return false;
+    }
+
+    const std::string& filename = info.filepath;
+    MeshMetaData& metaData = resourceDict_.at(filename);
+    auto indexPair = metaData.meshIndex;
+    int start = indexPair.first;
+    int end = indexPair.second;
+    bool objectSuccess;
+
+    std::vector<Magnum::Trade::MeshData3D*> meshGroup;
+
+    LOG(INFO) << "Accessing scene mesh start " << start << " end " << end;
+    for (int mesh_i = start; mesh_i < end; mesh_i++) {
+
+      // FRL Quad Mesh
+      if (info.type == AssetType::FRL_INSTANCE_MESH) {
+        FRLInstanceMeshData* frlMeshData = 
+            dynamic_cast<FRLInstanceMeshData*>(meshes_[mesh_i].get());
+        auto& meshData = frlMeshData->getMeshData();
+        if (!meshData) {
+          LOG(ERROR) << "Cannot load the mesh, skipping";
+          continue;
+        }
+        meshGroup.push_back(&meshData);
+      } 
+
+      // PLY Instance mesh
+      else if (info.type == AssetType::INSTANCE_MESH) {
+        GenericInstanceMeshData* insMeshData = 
+            dynamic_cast<GenericInstanceMeshData*>(meshes_[mesh_i].get());
+        quatf quatf = quatf::FromTwoVectors(info.frame.front(), geo::ESP_FRONT);
```

Comment:
```suggestion
        quatf R_world_frame = info.frame.rotationFrameToWorld();
```

Naming a quatf `quatf` is very scary :)